### ISSUE TITLE
Training - Testing Overlap Issue

### ISSUE THREAD

**Issue by Ajay5641** [fchest/EEG-AAD#7]
<img width="1474" height="934" alt="Image" src="https://github.com/user-attachments/assets/675a47e7-93c9-4aea-9871-086873608a5e" />

you mention that the data for each subject is split into an 8:1:1 ratio for training, validation, and testing. However, the preprocessing pipeline uses 1-second windows with a 0.5-second overlap, producing data of shape (6580, 128, 32) for a single subject.

My concern is that if the split is done after creating overlapping windows, then parts of the same time segment can end up in both training and testing. Because of the 50% overlap, a window in the test set may share half of its data with a window from the training set. This causes data leakage and can make the model look better than it actually is.

Could you clarify the exact procedure you followed to prevent this overlap-based data leakage? Did you generate the windows separately for each split, or did you apply another strategy to ensure isolation between training and testing segments?

Thank you.
